### PR TITLE
Slice operation support for ptabler

### PR DESCRIPTION
**Pull request by mzueva**
Required for MILAB-6013 batch processing API. Adds slice operation support to ptabler.

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds a `slice` operation to ptabler, exposing Polars' `LazyFrame.slice` as a first-class step across the TypeScript schema, Python executor, and Tengo SDK. The implementation is consistent and well-tested. As a minor aside, `LimitStep` was also added to the `index.ts` re-exports, which looks like a small pre-existing omission fixed alongside this work.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge; only finding is a P2 style suggestion about adding input validation to the Python model

All findings are P2 (style/best-practices). The core implementation is correct, tests cover the main edge cases, and the Tengo SDK already enforces the constraints at the call site.

lib/ptabler/software/src/ptabler/steps/slice.py — consider adding msgspec field validators for offset/length
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| lib/ptabler/software/src/ptabler/steps/slice.py | New PStep implementing Polars LazyFrame.slice; functionally correct but lacks input validation (offset ≥ 0, length > 0) enforced by the Tengo SDK layer |
| lib/ptabler/schema/src/basic_steps.ts | Adds well-documented SliceStep interface; constraints on offset/length are documented in JSDoc but not type-enforced (TypeScript limitation) |
| lib/ptabler/schema/src/index.ts | Adds SliceStep to PTablerStep union and re-exports; also adds the previously-missing LimitStep re-export |
| lib/ptabler/software/src/ptabler/steps/__init__.py | Correctly imports and exports Slice in both AnyPStep union and __all__ |
| lib/ptabler/software/src/test/slice_test.py | Good test coverage for basic range, out-of-range offset, length overflow, and zero-offset full-length cases |
| sdk/workflow-tengo/src/pt/index.lib.tengo | Adds slice() to the DataFrame fluent API with correct assertions (offset ≥ 0, length > 0) and step delegation |

</details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: lib/ptabler/software/src/ptabler/steps/slice.py
Line: 14-16

Comment:
**Missing input validation on `offset` and `length`**

The schema docs say `offset` is non-negative and the Tengo SDK explicitly asserts `length > 0`, but `Slice` has no corresponding guards. `msgspec.Struct` accepts any `int`, so a caller passing `offset=-1` would silently invoke Polars' from-end semantics (not what the contract describes), and `length=0` or a negative length could produce unexpected empty frames or a Polars runtime error. Other `PStep` implementations that carry semantic constraints (e.g. `Limit`) share this pattern; adding validators here would make the Python layer consistent with the Tengo layer.

```suggestion
    offset: Annotated[int, msgspec.Meta(ge=0)]
    length: Annotated[int, msgspec.Meta(gt=0)]
```
(requires `from typing import Annotated` at the top)

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["MILAB-6013 slice operation support for p..."](https://github.com/milaboratory/platforma/commit/1d33c79ce50f66e0b9dfe187622f05c093386d82) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=29119108)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: sdk/workflow-tengo/src/pt/index.lib.tengo [hunk 1]
```diff
@@ -1440,6 +1440,32 @@ _newDataFrame = func(parentWorkflow, dfName) {
...
+		 * @param offset {number} - Non-negative, 0-based starting row index.
+		 * @param length {number} - Positive number of rows to extract.
+		 * @returns {object} - A new DataFrame object.
+		 * @example
+		 *   batch0 := df.slice(0, 100)
+		 *   batch1 := df.slice(100, 100)
+		 */
+		slice: func(offset, length) {
+			ll.assert(is_int(offset) && offset >= 0, "slice offset must be a non-negative integer.")
+			ll.assert(is_int(length) && length > 0, "slice length must be a positive integer.")
```
[the remark this answers is not in the document]
Comment (mzueva) [reply sdk/workflow-tengo/src/pt/index.lib.tengo]: done

**Review by vadimpiven — APPROVED** [review]
